Financial Times.js

```diff
@@ -42,105 +36,94 @@
-ISSN			 : "0307-1766",
-publicationTitle : "Financial Times"
-}); 
- 
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('.search-item a.js-teaser-heading-link[href*="/content/"]');
+	for (let row of rows) {
+		let href = row.href;
+		let title = ZU.trimInternal(row.textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
 
-/** Articles */
-FW.Scraper({
-itemType         : 'newspaperArticle',
-detect           : FW.Xpath('//meta[@property="og:type" and @content="article" and //div[contains(@class, "fullstory")]//h1]'),
-title            : FW.Xpath('//div[contains(@class, "fullstory")]//h1').text().trim(),
-attachments      : [
-  {
-  url: FW.Url(),
-  title: "Financial Times Snapshot",
-  type: "text/html"}],
-creators         : FW.Xpath('//p[contains(@class, "byline")]/span').text().replace(/^By\s*/, "").remove(/ in .+/).split(/,| and /).cleanAuthor("author"),
-date             : FW.Xpath('//p[@id="publicationDate"]/span[@class="time"]').text(),
-abstractNote     : FW.Xpath('//meta[@name="description"]/@content').text(),
-ISSN			 : "0307-1766",
-issue			 : FW.Xpath('//div[contains(@class, "article-issue")]//div/a').text().trim(),
-publicationTitle : "Financial Times"
-});
- 
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (items) ZU.processDocuments(Object.keys(items), scrape);
+		});
+	}
+	else {
+		scrape(doc, url);
+	}
+}
 
- 
-FW.MultiScraper({
-itemType         : 'multiple',
-detect           : FW.Url().match(/\/search\?/),
-choices          : {
-  titles :  FW.Xpath('//li[contains(@class, "result")]/h3/a[contains(@href, "www.ft.com")]').text().trim(),
-  urls    :  FW.Xpath('//li[contains(@class, "result")]/h3/a[contains(@href, "www.ft.com")]').key("href")
+function scrape(doc, url) {
+	let item = new Zotero.Item('newspaperArticle');
+	let meta = JSON.parse(text('script[type="application/ld+json"]'));
+	if (meta['@type'] == 'WebSite') {
+		throw new Error("No article metadata (probably hit paywall)");
+	}
+	
+	item.title = meta.headline;
+	item.date = ZU.strToISO(meta.datePublished);
+	// abstracts and authors usually won't show up in tests - they're rendered
+	// client-side and only the first author makes it into the JSON-LD.
+	// https://github.com/Financial-Times/next-json-ld#example-markup-on-article-page-behind-paywall-as-seen-by-google-bot
+	item.abstractNote = meta.description
+		|| text('.o-topper__standfirst');
+	// something funky is going on with the JSON-LD authors, so we'll just
+	// parse from the HTML
+	item.creators = [...doc.querySelectorAll('a[data-trackable="author"]')]
+		.map(link => ZU.cleanAuthor(link.innerText, 'author', false));
+	if (meta.publisher) {
+		item.publicationTitle = meta.publisher.name;
```

Comment:
Is this ever not "Financial Times"? If missing, should it still be "Financial Times"? Any reason not to just hard-code?

Reply:
True. Done.